app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -329,89 +325,46 @@ fun MapScreen(
                 mapView.gestures.addOnMoveListener(moveListener)
                 // ======================= MOVE LISTENER =======================
 
-                // ======================= MARKERS CLICK =======================
-                markerAnnotationManager?.addClickListener(
-                    OnPointAnnotationClickListener { pointAnnotation ->
-                      mapViewModel.removePreviewCard()
-                      // get the data from the PointAnnotation and deserialize it
-                      val parkingData = pointAnnotation.getData()?.asJsonObject
-                      val parkingDeserialized = gson.fromJson(parkingData, Parking::class.java)
-                      // move the camera to the selected parking
-                      mapViewportState.setCameraOptions {
-                        center(parkingDeserialized.location.center)
+                // ======================= ANNOTATION CLICK LISTENER =======================
+                // Define a common click handler function
+                fun handleAnnotationClick(
```

Comment:
since it is a function maybe you can add the description for the parameters ?

Reply:
Sure, will add documentation on this